src/videopython/base/video.py

```diff
@@ -166,54 +170,77 @@ def split(self, frame_idx: int | None = None) -> tuple[Video, Video]:
+            filename: Name of the output video file. Generates random name if not provided.
+            format: Output format (default is 'mp4').
+
+        Returns:
+            Path to the saved video file.
         """
         if not self.is_loaded():
-            raise RuntimeError(f"Video is not loaded, cannot save!")
-
-        if filename is None:
-            filename = generate_random_name(suffix=".mp4")
-        filename = check_path(filename, dir_exists=True, suffix=".mp4")
-
-        ffmpeg_video_command = (
-            f"ffmpeg -loglevel error -y -framerate {self.fps} -f rawvideo -pix_fmt rgb24"
-            f" -s {self.metadata.width}x{self.metadata.height} "
-            f"-i pipe:0 -c:v libx264 -pix_fmt yuv420p {filename}"
-        )
-
-        ffmpeg_audio_command = (
-            f"ffmpeg -loglevel error -y -i {filename} -f s16le -acodec pcm_s16le "
-            f"-ar {self.audio.frame_rate} -ac {self.audio.channels} -i pipe:0 "
-            f"-c:v copy -c:a aac -strict experimental {filename}_temp.mp4"
-        )
+            raise RuntimeError("Video is not loaded, cannot save!")
 
-        try:
-            print("Saving frames to video...")
-            subprocess.run(
-                ffmpeg_video_command,
-                input=self.frames.tobytes(),
-                check=True,
-                shell=True,
+        # Check if the format is allowed
+        if format.lower() not in get_args(ALLOWED_VIDEO_FORMATS):
+            raise ValueError(
+                f"Unsupported format: {format}. Allowed formats are: {', '.join(get_args(ALLOWED_VIDEO_FORMATS))}"
             )
-        except subprocess.CalledProcessError as e:
-            print("Error saving frames to video!")
-            raise e
 
-        try:
-            print("Adding audio track...")
-            subprocess.run(ffmpeg_audio_command, input=self.audio.raw_data, check=True, shell=True)
-            Path(filename).unlink()
-            Path(filename + "_temp.mp4").rename(filename)
-        except subprocess.CalledProcessError as e:
-            print(f"Error adding audio track!")
-            raise e
-
-        print(f"Video saved into `{filename}`!")
-        return filename
+        if filename is None:
+            filename = Path(generate_random_name(suffix=f".{format}"))
+        else:
+            filename = Path(filename).with_suffix(f".{format}")
+            filename.parent.mkdir(parents=True, exist_ok=True)
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_dir_path = Path(temp_dir)
+
+            # Save frames as images
+            for i, frame in enumerate(self.frames):
+                frame_path = temp_dir_path / f"frame_{i:04d}.png"
+                cv2.imwrite(str(frame_path), cv2.cvtColor(frame, cv2.COLOR_RGB2BGR))
+
+            # Save audio to a temporary file
+            temp_audio = temp_dir_path / "temp_audio.wav"
+            self.audio.export(str(temp_audio), format="wav")
```

Comment:
Maybe we should export audio to aac if we use this format for saving video anyway?